src/main/java/iterator/RichIterator.java

```diff
@@ -298,7 +482,25 @@ default RichIterator<A> drop(int n) {
      * RichIterator.apply(1,2,3,4,5).take(20) // 1,2,3,4,5
      */
     default RichIterator<A> take(int n) {
-        throw new NotImplementedException();
+        return new RichIterator<A>() {
+            private int index = 0;
+
+            @Override
+            public boolean hasNext() {
+                return index < n && RichIterator.this.hasNext();
+            }
+
+            @Override
+            public A next() {
+                if (hasNext()) {
+                    index++;
+
+                    return RichIterator.this.next();
+                } else {
+                    throw new NoSuchElementException();
```

Comment:
In this case throwing the exception manually is justified because it can be thrown either by the wrapped `next` or by taking all the elements.

But it has an issue that if the standard is changed for some reason, this code also has to be changed.
Therefore a better solution is "returning" `empty().next()` instead which makes sense since the iterator is empty.

```suggestion
                    return empty().next();
```